infra/charts/turing/templates/_helpers.tpl

```diff
@@ -187,6 +195,7 @@ Sentry:
 {{ if .Values.turing.experimentEngines }}
 Experiment:
 {{ range $expEngine := .Values.turing.experimentEngines }}
+{{ if or (and (eq (toString $expEngine.name) "xp") (.Values).tags.xp ) (ne (toString $expEngine.name) "xp") }}
```

Comment:
Added the conditional check here to "glue" XP to Turing if it is enabled by checking if:
1. experiment engine name is "xp" **AND** tags.xp is enabled **OR** 
2. experiment engine name is not "xp"

Reply:
The file you've modified makes sense to me. But let's think about the conditions.. What it does is exclude the `xp` props if the XP tag has not been set. This is good. But, if the user set a new experiment engine in their values file (say 'test-engine') and XP is enabled, what is the behavior we will see?

The requirements are:
1. XP should be enabled if the `xp` tag is enabled
2. The user should be able to add other experiment engines.
3. The user should be able to override the experiment engine configs for XP (I didn't mention this earlier but you can imagine that this is required to override the hosts, even for our own internal deployment).

The helm [merge](https://helm.sh/docs/chart_template_guide/function_list/#mergeoverwrite-mustmergeoverwrite) operation will not combine the list values but rather, it will replace the destination's list value (the default values file) with that of the source (user-supplied values file). We need to add some more handling around merging of the experiment engine configurations to achieve the requirements and I think it's easiest done on the merged values - after the configs are merged, if (XP tag is enabled and there is no existing config for xp in the experiment engines dict, we can append the default experiment engine config for XP).

I think we also need to modify the `openApiSpecOverrides` in a similar manner.